toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexService.kt

```diff
@@ -279,8 +281,32 @@ class CratesLocalIndexService : PersistentStateComponent<CratesLocalIndexState>,
 private fun VFileEvent.pathEndsWith(suffix: String): Boolean =
     path.endsWith(suffix) || (this is VFilePropertyChangeEvent && oldPath.endsWith(suffix))
 
-data class CargoRegistryCrate(val versions: List<CargoRegistryCrateVersion>)
-data class CargoRegistryCrateVersion(val version: String, val isYanked: Boolean, val features: List<String>) {
+data class CargoRegistryCrate(val versions: List<CargoRegistryCrateVersion>) {
+    val latestVersion: CargoRegistryCrateVersion?
+        get() = versions.max()
+
+    val sortedVersions: List<CargoRegistryCrateVersion>
+        get() = versions.sorted()
+}
+
+data class CargoRegistryCrateVersion(
+    val versionString: String,
+    val isYanked: Boolean,
+    val features: List<String>
+) : Comparable<CargoRegistryCrateVersion> {
+
+    val version: Semver? by lazy {
+        try {
+            Semver(versionString, Semver.SemverType.NPM)
+        } catch (e: SemverException) {
+            null
+        }
+    }
+
+    override fun compareTo(other: CargoRegistryCrateVersion): Int {
+        return this.version?.compareTo(other.version) ?: return 0
```

Comment:
Note that currently object with `version == null` is considered **equal** to any other object. But I guess `null` should be considered **less** than any other objects. Also, calling `com.vdurmont.semver4j.Semver#compareTo` with `null` parameter value might be dangerous.

However, probably we should just ignore `CargoRegistryCrateVersion` with `version == null`. What do you think?
We can implement such logic by making a private constructor
```Kotlin
data class CargoRegistryCrateVersion private constructor(
    val version: Semver,
    val isYanked: Boolean,
    val features: List<String>
)
```
with a factory method to instantiate `CargoRegistryCrateVersion` from `versionString: String`